lib/spack/spack/user_environment.py

```diff
@@ -101,17 +115,6 @@ def environment_modifications_for_specs(
 
     # Apply view projections if any.
     if view:
-        prefix_to_prefix = {
-            s.prefix: view.get_projection_for_spec(s)
-            for s in reversed(topo_ordered)
-            if not s.external
-        }
-        # Avoid empty regex if all external
-        if not prefix_to_prefix:
-            return env
-        prefix_regex = re.compile("|".join(re.escape(p) for p in prefix_to_prefix.keys()))
-        for mod in env.env_modifications:
-            if isinstance(mod, environment.NameValueModifier):
-                mod.value = prefix_regex.sub(lambda m: prefix_to_prefix[m.group(0)], mod.value)
+        project_env_mods(*topo_ordered, view=view, env=env)
```

Comment:
I think this change loses the `reversed` from the deleted code. Is that an oversight or is it because it wasn't necessary?

Reply:
It shouldn't matter, cause externals are not projected to a view and spack prefixes are unique